sycl/include/sycl/vector_preview.hpp

```diff
@@ -411,8 +409,27 @@ class vec : public detail::vec_arith<DataT, NumElements> {
       T, /*->*/ T>::type;
   // clang-format on
 
-  // convertImpl can't be called with the same From and To types and therefore
-  // we need this version of convert which is mostly no-op.
+  // getValue should be able to operate on different underlying
+  // types: enum cl_float#N , builtin vector float#N, builtin type float.
+  constexpr auto getValue(int Index) const {
```

Comment:
I'm simply moving it closer to its uses. However, I think these `convert`-related details should be all moved into `vector_convert.hpp` in a later PR.